app/src/main/java/com/duckduckgo/app/browser/print/PrintJavascriptInterface.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (c) 2022 DuckDuckGo
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.duckduckgo.app.browser.print
+
+import android.webkit.JavascriptInterface
+
+class PrintJavascriptInterface(private val onPrintDetected: () -> Unit) {
+
+    @JavascriptInterface
+    fun print() {
+        onPrintDetected()
+    }
+
+    companion object {
+        const val JAVASCRIPT_INTERFACE_NAME = "Print"
+    }
```

Comment:
nit: is there any reason for this constant to be in this class as it's never access from here?

Reply:
This is the string name for the @JavascriptInterface. We create the instance outside this class giving it this `JAVASCRIPT_INTERFACE_NAME`, but then, if we want to call print function inside this interface, we'll need to use this interface name too. Therefore, I followed the same pattern you have with other JS interfaces (that I think it's good) and keep the interface name inside the companion object of the interface class although it is not called here. Does it make sense?

Reply:
Makes sense 👍